lib/model.js

```diff
@@ -1641,6 +1641,10 @@ Model.prototype.findOrCreate = function(options) {
     );
   }
 
+  if (this.sequelize.constructor.cls) {
+    options.transaction = this.sequelize.constructor.cls.get('transaction');
```

Comment:
Hmm although it's probably not possible to access the Sequelize class at this point so your method is probably the only way.